databricks/koalas/frame.py

```diff
@@ -6783,28 +6789,64 @@ def melt(self, id_vars=None, value_vars=None, var_name=None,
         1  b         B          3
         2  c         B          5
         """
+        column_index = self._internal.column_index
+
         if id_vars is None:
             id_vars = []
-        elif isinstance(id_vars, str):
-            id_vars = [(id_vars,)]
-        elif isinstance(id_vars, tuple):
-            if self._internal.column_index_level == 1:
-                id_vars = [idv if isinstance(idv, tuple) else (idv,) for idv in id_vars]
-            else:
-                raise ValueError('id_vars must be a list of tuples when columns are a MultiIndex')
         else:
-            id_vars = [idv if isinstance(idv, tuple) else (idv,) for idv in id_vars]
-
-        column_index = self._internal.column_index
+            if isinstance(id_vars, str):
+                id_vars = [(id_vars,)]
+            elif isinstance(id_vars, tuple):
+                if self._internal.column_index_level == 1:
+                    id_vars = [idv if isinstance(idv, tuple) else (idv,)
+                               for idv in id_vars]
+                else:
+                    raise ValueError('id_vars must be a list of tuples'
+                                     ' when columns are a MultiIndex')
+            else:
+                id_vars = [idv if isinstance(idv, tuple) else (idv,)
+                           for idv in id_vars]
+
+            non_existence_col = [idv for idv in id_vars if idv not in column_index]
+            if len(non_existence_col) != 0:
+                raveled_column_index = np.ravel(column_index)
+                missing = [nec for nec in np.ravel(non_existence_col)
+                           if nec not in raveled_column_index]
+                if len(missing) != 0:
+                    raise KeyError("The following 'id_vars' are not present"
+                                   " in the DataFrame: {}".format(missing))
+                else:
+                    raise KeyError("None of [{}] are in the [columns]"
+                                   .format(pd.MultiIndex.from_tuples(non_existence_col)))
```

Comment:
The error message is strange?

```
KeyError: "None of [MultiIndex(levels=[['X'], ['Y']],\n           codes=[[0], [0]])] are in the [columns]"
```

How about:

```diff
-                    raise KeyError("None of [{}] are in the [columns]"
-                                   .format(pd.MultiIndex.from_tuples(non_existence_col)))
+                    raise KeyError("None of {} are in the {}"
+                                   .format(non_existence_col, column_index))
```

then

```
KeyError: "None of [('X', 'Y')] are in the [('X', 'A'), ('X', 'B'), ('Y', 'C')]"
```

Reply:
I agree with you. 
I'll change error message the case of id_vars, value_vars.